fiddy/step.py

```diff
@@ -31,8 +36,15 @@ def dstep(
     #     index=dimension,
     #     value=size,
     # )
+    assert dimension < point.shape[0]
     array = np.zeros(point.shape)
     array[dimension] = size
+    if relative:
+        val = point[dimension]
+        if val == 0:
+            warnings.warn(f'point is zero valued in dimension {dimension},'
+                          f'resulting in zero step.')
```

Comment:
```suggestion
            warnings.warn(f'point is zero valued in dimension {dimension}, '
                          f'resulting in zero step.')
```

I am wondering if this could be handled better. Provide a fall-back absolute step in case the step would be zero? Interpret the supposedly relative step as absolut? Fine as is for now.